packages/apollo-link-http/src/httpLink.ts

```diff
@@ -144,8 +144,8 @@ export const createHttpLink = (linkOptions: HttpLink.Options = {}) => {
           return result;
         })
         .catch(err => {
-          // fetch was cancelled so its already been cleaned up in the unsubscribe
-          if (err.name === 'AbortError') return;
+          // exit if fetch was cancelled from the observer unsubscribing
+          if (err.name === 'AbortError' && observer.closed) return;
```

Comment:
Think you don't even need to check what type of error it is. If nobody is listening (`observer.closed`), you can return here.

Reply:
In fact, you shouldn't care what type of error it is. As long as someone is subscribed, they need to find out about this error, otherwise you're just leaving them hanging.

PS: As you probably figured out, the query deduplicator never finding out that the fetch was aborted is what's causing subsequent requests to hang.